sycl/source/platform.cpp

```diff
@@ -56,11 +56,37 @@ std::vector<platform> platform::get_platforms() {
 
 backend platform::get_backend() const noexcept { return impl->getBackend(); }
 
+#ifdef __INTEL_PREVIEW_BREAKING_CHANGES
+template <typename Param>
+typename detail::is_platform_info_desc<Param>::return_type
+platform::get_info_internal() const {
+  return impl->get_info<Param>();
+}
+
+void platform::get_platform_info(detail::string_view &Type,
+                                 detail::string &InfoStr) const {
+  std::string Info;
+  if (Type == typeid(info::platform::name).name()) {
+    Info = impl->get_info<info::platform::name>();
+  } else if (Type == typeid(info::platform::vendor).name()) {
+    Info = impl->get_info<info::platform::vendor>();
+  } else if (Type == typeid(info::platform::version).name()) {
+    Info = impl->get_info<info::platform::version>();
+  } else if (Type == typeid(info::platform::profile).name()) {
+    Info = impl->get_info<info::platform::profile>();
+  } else {
+    throw sycl::invalid_parameter_error("unsupported device info requested",
+                                        PI_ERROR_INVALID_OPERATION);
+  }
+  InfoStr = Info;
+}
```

Comment:
Same suggestions as above

Reply:
Please elaborate the rationale of  your suggestion.

Reply:
Current implementation needs a translation from a type template argument to an ID to a string, which is used as a key in this `get_platform_info` function. However, the function does exactly the same transformation to end up comparing the resulting strings. Using a template function would save us that whole translation process and simplify control flow.

Reply:
I see.
This is also a special case where we need to handle `std::string` is crossing the ABI boundary.
The template parameter itself is indeed std::string, and the header's template param and libsycl's template param do not speak the same language.
So, I had to specialize those string template parameter cases in the header file (pre-C++11 ABI) and marshall the template type into a detail::string parameter to the libsycl(post-C++11 ABI). 
So, template for the std::string parameter types won't work across different ABIs.
I will fix the reference type to pass-by-value.